IAM samples using "allUsers" principal are now blocked on CI project

The resources inside our CI project are no longer allowed to grant roles to the `allUsers` principal

E.g., from [this failed build](https://source.cloud.google.com/results/invocations/4155fc80-0b31-419f-b5bd-732fa72b5682/targets/cloud-devrel%2Fclient-libraries%2Fpython%2Fgoogleapis%2Fpython-pubsub%2Fsamples%2Fpython3.6%2Fpresubmit/log):

```python
____________________________ test_set_topic_policy _____________________________

args = (resource: "projects/python-docs-samples-tests/topics/iam-test-topic-08d0a54da5e04194b85c68c21b1c1782"
policy {
  etag...ers: "allUsers"
  }
  bindings {
    role: "roles/pubsub.publisher"
    members: "group:cloud-logs@google.com"
  }
}
,)
kwargs = {'metadata': [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/topics/iam-test-topic-08d0a54da5e04194b85c68c21b1c1782'), ('x-goog-api-client', 'gl-python/3.6.15 grpc/1.41.1 gax/2.1.1 gccl/2.8.0')]}

    @functools.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
>           return callable_(*args, **kwargs)

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:66:
...
args = (resource: "projects/python-docs-samples-tests/topics/iam-test-topic-08d0a54da5e04194b85c68c21b1c1782"
policy {
  etag...ers: "allUsers"
  }
  bindings {
    role: "roles/pubsub.publisher"
    members: "group:cloud-logs@google.com"
  }
}
,)
kwargs = {'metadata': [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/topics/iam-test-topic-08d0a54da5e04194b85c68c21b1c1782'), ('x-goog-api-client', 'gl-python/3.6.15 grpc/1.41.1 gax/2.1.1 gccl/2.8.0')]}

    @functools.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
            return callable_(*args, **kwargs)
        except grpc.RpcError as exc:
>           raise exceptions.from_grpc_error(exc) from exc
E           google.api_core.exceptions.FailedPrecondition: 400 One or more users named in the policy do not belong to a permitted customer.

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:68: FailedPrecondition
_________________________ test_set_subscription_policy _________________________

args = (resource: "projects/python-docs-samples-tests/subscriptions/iam-test-subscription-08d0a54da5e04194b85c68c21b1c1782"
p..."
    members: "allUsers"
  }
  bindings {
    role: "roles/editor"
    members: "group:cloud-logs@google.com"
  }
}
,)
kwargs = {'metadata': [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/subscriptions/iam-test-subscription-08d0a54da5e04194b85c68c21b1c1782'), ('x-goog-api-client', 'gl-python/3.6.15 grpc/1.41.1 gax/2.1.1 gccl/2.8.0')]}

    @functools.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
>           return callable_(*args, **kwargs)

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:66:
...
args = (resource: "projects/python-docs-samples-tests/subscriptions/iam-test-subscription-08d0a54da5e04194b85c68c21b1c1782"
p..."
    members: "allUsers"
  }
  bindings {
    role: "roles/editor"
    members: "group:cloud-logs@google.com"
  }
}
,)
kwargs = {'metadata': [('x-goog-request-params', 'resource=projects/python-docs-samples-tests/subscriptions/iam-test-subscription-08d0a54da5e04194b85c68c21b1c1782'), ('x-goog-api-client', 'gl-python/3.6.15 grpc/1.41.1 gax/2.1.1 gccl/2.8.0')]}

    @functools.wraps(callable_)
    def error_remapped_callable(*args, **kwargs):
        try:
            return callable_(*args, **kwargs)
        except grpc.RpcError as exc:
>           raise exceptions.from_grpc_error(exc) from exc
E           google.api_core.exceptions.FailedPrecondition: 400 One or more users named in the policy do not belong to a permitted customer.

.nox/py-3-6/lib/python3.6/site-packages/google/api_core/grpc_helpers.py:68: FailedPrecondition
```

--- a/samples/snippets/iam.py
+++ b/samples/snippets/iam.py
@@ -81,7 +81,7 @@ def set_topic_policy(project_id: str, topic_id: str) -> None:
     policy = client.get_iam_policy(request={"resource": topic_path})
 
     # Add all users as viewers.
-    policy.bindings.add(role="roles/pubsub.viewer", members=["allUsers"])
+    policy.bindings.add(role="roles/pubsub.viewer", members=["domain:google.com"])
 
     # Add a group as a publisher.
     policy.bindings.add(
@@ -110,7 +110,7 @@ def set_subscription_policy(project_id: str, subscription_id: str) -> None:
     policy = client.get_iam_policy(request={"resource": subscription_path})
 
     # Add all users as viewers.
-    policy.bindings.add(role="roles/pubsub.viewer", members=["allUsers"])
+    policy.bindings.add(role="roles/pubsub.viewer", members=["domain:google.com"])
 
     # Add a group as an editor.
     policy.bindings.add(role="roles/editor", members=["group:cloud-logs@google.com"])

--- a/samples/snippets/iam_test.py
+++ b/samples/snippets/iam_test.py
@@ -98,7 +98,7 @@ def test_set_topic_policy(
     iam.set_topic_policy(PROJECT_ID, TOPIC_ID)
     policy = publisher_client.get_iam_policy(request={"resource": topic_path})
     assert "roles/pubsub.publisher" in str(policy)
-    assert "allUsers" in str(policy)
+    assert "domain:google.com" in str(policy)
 
 
 def test_set_subscription_policy(
@@ -107,7 +107,7 @@ def test_set_subscription_policy(
     iam.set_subscription_policy(PROJECT_ID, SUBSCRIPTION_ID)
     policy = subscriber_client.get_iam_policy(request={"resource": subscription_path})
     assert "roles/pubsub.viewer" in str(policy)
-    assert "allUsers" in str(policy)
+    assert "domain:google.com" in str(policy)
 
 
 def test_check_topic_permissions(topic_path: str, capsys: CaptureFixture) -> None:

